fhir-resource-tooling/src/main/java/com/ibm/cohort/tooling/fhir/ValueSetImporter.java

```diff
@@ -73,9 +105,35 @@ static void runWithArgs(String[] args, PrintStream out) throws IOException {
 			for (String arg : arguments.spreadsheets) {
 				try (InputStream is = new FileInputStream(arg)) {
 					ValueSetArtifact artifact = ValueSetUtil.createArtifact(is, codeSystemMappings);
-					String retVal = ValueSetUtil.importArtifact(client, artifact, arguments.overrideValueSets);
-					if(retVal == null){
-						logger.error("Value set already exists! Please provide the override option if you would like to override this value set.");
+					
+					//only import the value set to fhir server if we are not writing the value set to file system
+					if(arguments.fileSystemOutputPath == null) {
+						String retVal = ValueSetUtil.importArtifact(client, artifact, arguments.overrideValueSets);
+						if(retVal == null){
+							logger.error("Value set already exists! Please provide the override option if you would like to override this value set.");
+						}
+					}else {
+						//write value set to file system
+						ValueSet vs = artifact.getFhirResource();
+						
+						//If the valueset id contains urn:oid, remove it to make a valid filename
+						String valueSetId = vs.getId().startsWith("urn:oid:") ? vs.getId().replace("urn:oid:", "") : vs.getId();						
+						String vsFileName = valueSetId + "." +arguments.filesystemOutputFormat.toLowerCase();
+						
+						//create the output dir if it doesn't exist
+						File outputDir = new File(arguments.fileSystemOutputPath);
+						if(!outputDir.exists()) {
+							outputDir.mkdir();
+						}
+						
+						//write to xml or json format
+						BufferedWriter writer = new BufferedWriter(new FileWriter(arguments.fileSystemOutputPath + System.getProperty("file.separator")+vsFileName));
```

Comment:
This should be in a try-with-close block (e.g. try( BufferedWriter writer = new .... ) { /* do something */ }

Reply:
On a separate note, I wonder if we should introduce a hadoop dependency here or if local filesystem is ok? The ultimate destination for the files for most consumers is going to be an S3 path.